src/qtism/data/content/interactions/ExtendedTextInteraction.php

```diff
@@ -505,7 +507,13 @@ public function getResponseValidityConstraint(): ResponseValidityConstraint
             $this->getResponseIdentifier(),
             $this->getMinStrings(),
             ($this->hasMaxStrings() === false) ? 0 : $this->getMaxStrings(),
-            $this->getPatternMask()
+            $this->getPatternMask(),
+            [
+                'qtiClassName' => $this->getQtiClassName(),
+                'options' => [
+                    'format' => $this->getFormat(),
+                ],
+            ]
```

Comment:
What if instead of loosely-typed `extraData` we provisioned a reference to the whole instance of the current `Interaction` here?
```suggestion
            $this
```

Instead of `extraData` property on the constraint level, you can then have `private ?Interaction = null`.

Do you foresee any problems with that? If so, or if you don't think it makes any sense, ignore this comment.

Reply:
Yes, it was my first idea, but we have several problems with it on serialized and unserialized actions for the final session. It is not super practice, but it is better to keep here simple data.